temporian/core/operators/lag.py

```diff
@@ -84,21 +86,67 @@ def build_op_definition(cls) -> pb.OperatorDef:
 operator_lib.register_operator(LagOperator)
 
 
-def lag(event: Event, duration: Duration) -> Event:
-    if duration <= 0:
-        raise ValueError("duration must greater than zero")
+def _base_lag(
```

Comment:
This function is used both for lag and leak. This is the lag/leak file. Therefore, it is probably okay to call this method: "_implementation".